plugins/dicom_viewer/server/__init__.py

```diff
@@ -195,9 +196,44 @@ def _uploadHandler(event):
     events.trigger('dicom_viewer.upload.success')
 
 
+def dicomSubstringSearchHandler(query, types, user=None, level=None, limit=0, offset=0):
+    """
+    Provide a substring search on both keys and values.
+    """
+    if types != ['item']:
+        raise TypeError('The dicom search is only able to search in Item.')
+
+    # The insensitive Case is not perfect
+    # We will need to search into object value as improvment.
+    jsQuery = """
+        return Object.keys(obj.dicom.meta).some(
+            function(key) {
+                return (key.toLowerCase().indexOf('%(key)s'.toLowerCase()) !== -1)  ||
+                    obj.dicom.meta[key].toString().toLowerCase()
+                        .indexOf('%(value)s'.toLowerCase()) !== -1;
+            })
+        """ % {'key': query, 'value': query}
```

Comment:
Right now, there's no need for the `return`, since `$where` can accept a simple JavaScript expression.

However, I think you should wrap this whole `jsQuery` into a JS function, so we can define some intermediate variables for performance. Specifically, I think you should define variables for at least:
* `'%(key)s'.toLowerCase()`
* `'%(value)s'.toLowerCase()`
* `obj.dicom.meta`